File: Ruby/problem038.rb
#! /usr/bin/env ruby
=begin
Problem 38 - Project Euler
http://projecteuler.net/index.php?section=problems&id=38
=end

class Integer
    def pandigital_9digits?
        self.to_s.chars.map(&:to_i).push(0).uniq.size == 10
    end
end

puts (9183..9999).map {|i| [i, i*2].map(&:to_s).join.to_i }
                .select(&:pandigital_9digits?)
                .max

File: Kotlin/problem038.kt
// Problem 38 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=38
fun isPandigital9digits(x: Int): Boolean {
  return (x.toString() + "0").toSet().count() == 10
}

fun main() {
  val maxp =
      (9183..9999)
          .map { p -> (p.toString() + (p * 2).toString()).toInt() }
          .filter { isPandigital9digits(it) }
          .max()
  println(maxp)
}
